### ISSUE TITLE
Docker events engine broken on newer docker.py 

### ISSUE THREAD

**Issue by The-Loeki** [saltstack/salt#43729]
### Description of Issue/Question
With `docker-py` 2.0+ some stuffs changed around. This seems fixed in the `dockermod._get_client`, which however isn't used in the Engine.

### Setup
Try enabling the docker events engine w/up-to-date `docker-py`.

### Related
#39216 

@terminalmage 
Move the `get_client` into `utils` and reuse it on both the engine & the module sound like a plan? Or overload import through `salt.compat`?

### Versions report
```
Salt Version:
           Salt: 2017.7.1-1185-g759c3d5
 
Dependency Versions:
      docker-py: 2.5.1

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 4.10.8-fedora-1
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

### Example output
```
2017-09-25 22:22:14,613 [salt.utils.process:336 ][DEBUG   ][4650] Started 'salt.engines.Engine' with pid 24840
2017-09-25 22:22:14,644 [salt.utils.lazy  :97  ][DEBUG   ][24840] LazyLoaded docker_events.start
2017-09-25 22:22:14,646 [salt.engines     :130 ][CRITICAL][24840] Engine <salt.loader.LazyLoader object at 0x28467d0> could not be started! Error: 'module' object has no attribute 'Client'
```

**Comment by Ch3LL** [comment]
Definitely will need @terminalmage to speak up here since he has recently done alot of work here. Do you like the idea of utils @terminalmage for reuse?

**Comment by terminalmage** [comment]
Fixed in https://github.com/saltstack/salt/pull/43733.

**Comment by terminalmage** [comment]
@The-Loeki once this is merged into 2017.7 I will abstract client spawning into salt.utils.docker (which was added for that release). In the meantime, you can get this to work if you `pip uninstall docker` and `pip install docker-py` (which will install the legacy docker-py module that still uses the `docker.Client` attribute.

**Comment by The-Loeki** [comment]
np, tnx for the quick resolution :)
